test/box-luatest/gh_4264_recursive_trigger_test.lua

```diff
@@ -41,3 +41,24 @@ g.test_recursive_trigger_invocation = function(cg)
     t.assert_equals(order, {11, 21, 31, 32, 22, 12},
                     "Correct recursive trigger invocation")
 end
+
+g.test_trigger_clear_from_trigger = function(cg)
+    local order = cg.server:exec(function()
+        local order = {}
+        local s = box.space.test
+        rawset(_G, 'f2', function()
```

Comment:
Why do you need `f2` to be global? AFAIU it could just as well be local and captured in the functions:

```Lua
local f1, f2
f1 = ...
f2 = ...
```

Reply:
That's my oversight. I noticed that with `local f1 = function() ... `   `f1` can't be captured by the function and thought it must be global to work.
```
local f1
f1 = function() ...
```
works just fine.
<details><summary>diff</summary>

```diff
diff --git a/test/box-luatest/gh_4264_recursive_trigger_test.lua b/test/box-luatest/gh_4264_recursive_t>
index 4bfb10f81..a61a774a3 100644
--- a/test/box-luatest/gh_4264_recursive_trigger_test.lua
+++ b/test/box-luatest/gh_4264_recursive_trigger_test.lua
@@ -46,16 +46,17 @@ g.test_trigger_clear_from_trigger = function(cg)
     local order = cg.server:exec(function()
         local order = {}
         local s = box.space.test
-        rawset(_G, 'f2', function()
+        local f2 = function()
             table.insert(order, 2)
-        end)
-        rawset(_G, 'f1', function()
+        end
+        local f1
+        f1 = function()
             table.insert(order, 1)
-            s:on_replace(nil, _G.f2)
-            s:on_replace(nil, _G.f1)
-        end)
-        s:on_replace(_G.f2)
-        s:on_replace(_G.f1)
+            s:on_replace(nil, f2)
+            s:on_replace(nil, f1)
+        end
+        s:on_replace(f2)
+        s:on_replace(f1)
         s:replace{2}
         return order
     end)
```

</details>